packages/core/src/contrib/__tests__/Constraints.test.ts

```diff
@@ -157,23 +187,44 @@ describe("simple constraint", () => {
 });
 
 describe("general constraints", () => {
+  const expectSatified = (x: VarAD) => {
+    expect(numOf(x)).toBeLessThanOrEqual(1e-5);
+  };
+  const expectJustSatified = (x: VarAD) => {
+    expect(numOf(x)).toBeCloseTo(0, 5);
+  };
+  const expectNotSatisfied = (x: VarAD) => {
+    expect(numOf(x)).toBeGreaterThan(5);
+  };
+
   // Overlapping shapes
   it.each([
```

Comment:
Just wanted to say it: these tests are great! We've been talking about adding them but never did it.